### PR TITLE
Patch 1

### PR DESCRIPTION
**Pull request by doekia**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.6.1.x
| Description?  | [describe what is fixed by your PR]
| Type?         | bug fix
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | no
| How to test?  | [describe how to test]

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/8475)

### CODE REVIEW

**Comment by LittleBigDev** [comment]
Hello @doekia, I set back the PR contribution table. Could you please fill the missing descriptions ?
Also, please set a more explicit title to the PR.
Thanks by advance

**Comment by LittleBigDev** [comment]
Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[js/admin/products.js, line 180 at r1](https://reviewable.io:443/reviews/prestashop/prestashop/8475#-KyM6J6X03ez9Cp_QpJK:-KyM6J6Y97OabkudhNtJ:bhggjc7) ([raw file](https://github.com/prestashop/prestashop/blob/2c7e04d62070e1b65aa1770fd290421edb353f4c/js/admin/products.js#L180)):*
> ```JavaScript
> 						current_tab = $(request.responseText).filter('.product-tab').attr('id').replace('product-', '');
> 					}
> 					} catch(e){};
> ```

What kind of exception are you masking here ? I wonder if we can fix the exception instead of hiding it
Also, indentation problem.

---

*[js/jquery/plugins/alerts/jquery.alerts.css, line 49 at r1](https://reviewable.io:443/reviews/prestashop/prestashop/8475#-KyM5QXFD72eqc62Xjra:-KyM5QXFD72eqc62Xjrb:bsp7j5i) ([raw file](https://github.com/prestashop/prestashop/blob/2c7e04d62070e1b65aa1770fd290421edb353f4c/js/jquery/plugins/alerts/jquery.alerts.css#L49)):*
> ```CSS
> 	padding-left: 48px;
> 	max-height: 350px;
> 	overflow: auto;
> ```

I'm afraid you cannot add style in this file. This is part of an external jQuery module.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/prestashop/prestashop/8475)*

**Review by LittleBigDev — CHANGES_REQUESTED** [review]
Thanks @doekia for your contribution.
Some changes / additional info are requested though. See comments above.

**Comment by doekia** [comment]
Hum, be more explicit?
This code is fixing the existing crapy code. 
Why crapy? 
The actual code is: "when there is an error, lets crash the entire controller page"
And the external jQuery module (not updated since age) without the fix, froze the entire page since the <OK> button is unreachable.

Do whatever you want, keep you bugs if you wish - not my problem

**Comment by Quetzacoalt91** [comment]
Is there a reason to feel triggered by a response? Bringing a contribution doesn't mean you can spit on everything.

**Comment by mickaelandrieu** [comment]
> Do whatever you want, keep you bugs if you wish - not my problem

Then don't contribute!

For me, it's :-1:  no matter if you're right or not, this kind of behavior must be forbidden.

**Review by mickaelandrieu — CHANGES_REQUESTED** [review]
This need a better implementation (never hide an error!), and a better behavior (apologies, maybe?)

**Comment by vincentbz** [comment]
Hi @doekia 

I don't get the point. You make the effort to contribute to the Prestashop project, which is nice of you.
There is a lack of information in the PR, which are asked to make everything easier (PR qualification and understanding, search and so on).
@LittleBigDev  was not rude, just asking politely these information so that the PR respects the rules. If nobody can formalize the PR, it will start again as it was before, a mess from every side.

Regards

**Comment by doekia** [comment]
I do not mean to be rude to any of you as individual. 

I am tired of those PR posted to become buried. This PR is 2 lines, and I truly don't know how to qualify the fix.
It fixed a code written to handle error condition that not only do not do the job but send the page to oblivion - any one prefer this phrasing.

**Comment by LittleBigDev** [comment]
@doekia
About your salty comments, I'll just let it go because I know what it feels. I've been there too in the past.
About my review, I confirm my comments and requests. And they are totally explicit : I refuse to hide an error. I want to fix it instead.
About the buried PRs, you need to know we are working hard on getting back on our feet. No spiel. You'll see soon.
About the "not my problem", well... This IS your problem. You even contributed trying to fix it. Let's swallow our pride, and work together. Ok ?

**Comment by doekia** [comment]
It is an error handler code, I don't think it will be time spending to make it cope with - per design an unexpected situation. I presume the request.responseText will never contains any tab information, but if it has, let's try to have a nice handler.
The patch is exactly following your though, display as much as possible of the error, but (reason of alert.css fix) not at the cost of rendering the the interface frozen.

When I say "not my problem", I mean I have fixed the code for my customers, it is now part of my toolbox. I am proposing it to you. That it been accepted, rejected, postponed is not really a concern to me. The main reason for such a call was the mention about "not been authorised to change the css". Sure it is not PS code, nevertheless it is buggy.

**Comment by Eolia** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[js/admin/products.js, line 180 at r1](https://reviewable.io:443/reviews/prestashop/prestashop/8475#-KyM6J6X03ez9Cp_QpJK:-L-uVegm0dQWbMksA456:b-9t40ix) ([raw file](https://github.com/prestashop/prestashop/blob/2c7e04d62070e1b65aa1770fd290421edb353f4c/js/admin/products.js#L180)):*
<details><summary><i>Previously, LittleBigDev (Bastien Bieri) wrote…</i></summary><blockquote>

What kind of exception are you masking here ? I wonder if we can fix the exception instead of hiding it
Also, indentation problem.
</blockquote></details>

Seriously ?
The goal of try / catch is not to hide but to avoid the crash of the buttons on the page.
It also provides an answer about the error, if it exists.
The reasons for a crash can be multiple and Prestashop users are not all experts in code.
Showing an error (and not hiding it !) Helps to find the origin.

It's incredible that we have to explain this to you ...
We simply want to improve the user experience and unlock situations. Do you understand that?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/prestashop/prestashop/8475)*

**Comment by LittleBigDev** [comment]
Did you EVEN have a look at the PR ?
```javascript
} catch(e){};
```
Display **what** to **who** ?
You are right, this would be incredible if you had to explain this to me. A wise head makes a close mouth.

**Comment by Eolia** [comment]
Incredible...

> It also provides an answer about the error, **if it exists**.

But the most important thing is to not block the page.

**Comment by LittleBigDev** [comment]
Looks like I have to explain to you that an empty `catch` block will swallow the exception in nothingness, hiding the crash to the rest of the world.
Rejected, period. Stop trolling (I hope this is nothing but troll).

File: js/admin/products.js [hunk 1]
```diff
@@ -173,11 +173,13 @@ function ProductTabsManager(){
 
 				if ((in_array(status, wrong_statuses) || in_array(request.status, wrong_status_code)) && !self.page_reloading) {
 					var current_tab = '';
+					try {
 					if (request.responseText !== 'undefined' && request.responseText && request.responseText.length) {
 						current_tab = $(request.responseText).filter('.product-tab').attr('id').replace('product-', '');
 					}
+					} catch(e){};
```
Comment (mickaelandrieu) [inline js/admin/products.js:180]: nope!

**Comment by mickaelandrieu** [comment]
Hello,

thanks for your contribution.

Sadly, we can't accept it and you don't want to update it.

Regards,

**Comment by doekia** [comment]
The catch does nothing because the only error that could trigger here is that the tab name does not exists. It does not exists since the messageText contains the real error rather than a (when no error occur) properly formatted content. And the real error is what we want to display, not the fact that an error occured trying to process a normal message.

**Comment by doekia** [comment]
Feel free to switch to French language should you have difficulties understanding the stakes here - may be wiser than stating contribution as trolling - that is rather insulting

**Pull request closed without merging**